packages/utils/src/Pending.ts

```diff
@@ -43,15 +53,15 @@ class Pending {
     }
   }
 
-  resolve(promise) {
+  resolve(promise: CancelablePromise<unknown>): void {
     this.remove(promise);
     this.resolved.push(promise);
```

Comment:
Not part of this change, but I don't think this is actually resolving the promise. Just removes it from tracking array and then says it's resolved even if it isn't